src/order/ideal.lean

```diff
@@ -85,6 +90,18 @@ lemma inter_nonempty [preorder P] [ideal_inter_nonempty P] :
 ∀ (I J : ideal P), (I.carrier ∩ J.carrier).nonempty :=
 ideal_inter_nonempty.inter_nonempty
 
+/-- A preorder `P` has the `ideal_Inter_nonempty` property if the
+    intersection of all ideals is nonempty.
+    Most importantly, a `semilattice_sup` preorder with this property
+    satisfies that its ideal poset is a complete lattice.
+-/
+class ideal_Inter_nonempty (P) [preorder P] : Prop :=
+(Inter_nonempty : (⋂ (I : ideal P), I.carrier).nonempty)
```

Comment:
```suggestion
(Inter_nonempty : (⋂ (I : ideal P), (I : set P)).nonempty)
```

Reply:
This actually doesn't work because `has_coe` is only defined later (in the preorder section).
Should I keep this with `I.carrier`, move has_coe earlier or move the definition to the preorder section?

Reply:
Another possibility (probably worse) is to call the field `Inter_nonempty'` and then define `ideal_Inter_nonempty.Inter_nonempty` later. But sure, if you can move it to somewhere where it does work then this seems like the nicest solution.

Reply:
Looking more carefully, why not just move `ideal_Inter_nonempty` and `Inter_nonempty` to just before you start to use them on line 170 or so?